auth/authenticate.js

```diff
@@ -1,29 +1,41 @@
 const jwt = require('jsonwebtoken');
 
-const jwtKey =
-  process.env.JWT_SECRET ||
-  'add a .env file to root of project with the JWT_SECRET variable';
+const jwtKey = process.env.JWT_SECRET || 'add a .env file to root of project with the JWT_SECRET variable';
 
 // quickly see what this file exports
 module.exports = {
-  authenticate,
+	authenticate,
+	makeToken
 };
 
 // implementation details
 function authenticate(req, res, next) {
-  const token = req.get('Authorization');
+	const token = req.headers.authorization;
 
-  if (token) {
-    jwt.verify(token, jwtKey, (err, decoded) => {
-      if (err) return res.status(401).json(err);
-
-      req.decoded = decoded;
+	if (token) {
+		jwt.verify(token, jwtKey, (err, decoded) => {
+			if (err) {
+				return res.status(401).json({ message: 'Not authorized' });
+			} else {
+				req.decoded = decoded;
+				next();
+			}
+		});
+	} else {
+		return res.status(401).json({
+			error: 'No token provided, must be set on the Authorization Header'
+		});
+	}
+}
 
-      next();
-    });
-  } else {
-    return res.status(401).json({
-      error: 'No token provided, must be set on the Authorization Header',
-    });
-  }
+function makeToken(user) {
+	const payload = {
+		subject: user.id,
+		username: user.username
+	};
+	const options = {
+		expiresIn: '1h'
```

Comment:
import duration from env file do you can switch between a production version of this value and the devlopment version